lxd/daemon.go

```diff
@@ -1803,14 +1803,17 @@ func (d *Daemon) setupRBACServer(rbacURL string, rbacKey string, rbacExpiry int6
 	var err error
 
 	if d.authorizer != nil {
-		d.authorizer.StopStatusCheck()
+		err := d.authorizer.StopService(d.shutdownCtx)
+		if err != nil {
+			logger.Error("Failed to stop authorizer service", logger.Ctx{"error": err})
```

Comment:
We should return an error here rather than logging right? Otherwise we can get into an inconsistent state?

Reply:
Or do we also need to fallback to TLS in that case?